packages/openapi3/src/openapi.ts

```diff
@@ -887,6 +919,114 @@ function createOAPIEmitter(program: Program, options: OpenAPIEmitterOptions) {
     }
   }
 
+  function validateDiscriminator(discriminator: any, childModels: ModelType[]): boolean {
```

Comment:
As before, I think some of this should be simplified by having more functionality in the extension implementing the decorator, although special validations will occur for many emitters, on top of basic validation (no system would allow a single key to map to multiple types, for example..